src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.cs

```diff
@@ -150,12 +195,7 @@ public void AnalyzeOperation(OperationAnalysisContext context)
 
             public void OperationBlockEndAction(OperationBlockAnalysisContext context)
             {
-                // Report diagnostics for unused parameters.
-                foreach (IParameterSymbol parameter in _unusedParameters)
-                {
-                    Diagnostic diagnostic = Diagnostic.Create(Rule, parameter.Locations[0], parameter.Name, parameter.ContainingSymbol.Name);
-                    context.ReportDiagnostic(diagnostic);
-                }
+                _finalUnusedParameters[_method] = _unusedParameters;
```

Comment:
Can you switch this to .Add invocation (stronger assertion that we are overwriting any existing entry)?